src/main/cpp/util/file_posix.cc

```diff
@@ -444,15 +444,16 @@ void ForEachDirectoryEntry(const string &path,
 
     string filename(blaze_util::JoinPath(path, ent->d_name));
     bool is_directory;
-    if (ent->d_type == DT_UNKNOWN) {
-      struct stat buf;
-      if (lstat(filename.c_str(), &buf) == -1) {
-        die(blaze_exit_code::INTERNAL_ERROR, "stat failed: %s",
-            GetLastErrorString().c_str());
-      }
-      is_directory = S_ISDIR(buf.st_mode);
-    } else {
+    // 'd_type' field isn't part of the POSIX spec.
+    #ifdef _DIRENT_HAVE_D_TYPE
```

Comment:
Same as in https://github.com/bazelbuild/bazel/pull/4969 -- is the motivation here to compile Bazel on some POSIX system or just theoretical POSIX compatibility?